xarray/tests/test_duck_array_ops.py

```diff
@@ -672,17 +674,56 @@ def test_datetime_to_numeric_datetime64():
 
 @requires_cftime
 def test_datetime_to_numeric_cftime():
-    times = cftime_range("2000", periods=5, freq="7D").values
-    result = duck_array_ops.datetime_to_numeric(times, datetime_unit="h")
+    times = cftime_range("2000", periods=5, freq="7D", calendar="standard").values
+    result = duck_array_ops.datetime_to_numeric(times, datetime_unit="h", dtype=int)
     expected = 24 * np.arange(0, 35, 7)
     np.testing.assert_array_equal(result, expected)
 
     offset = times[1]
-    result = duck_array_ops.datetime_to_numeric(times, offset=offset, datetime_unit="h")
+    result = duck_array_ops.datetime_to_numeric(
+        times, offset=offset, datetime_unit="h", dtype=int
+    )
     expected = 24 * np.arange(-7, 28, 7)
     np.testing.assert_array_equal(result, expected)
 
     dtype = np.float32
     result = duck_array_ops.datetime_to_numeric(times, datetime_unit="h", dtype=dtype)
     expected = 24 * np.arange(0, 35, 7).astype(dtype)
     np.testing.assert_array_equal(result, expected)
+
+
+@requires_cftime
+@pytest.mark.xfail(
+    np.__version__ < "1.17", reason="Limited Numpy support for timedelta conversion."
+)
+def test_datetime_to_numeric_overflow():
+    import cftime
+
+    times = pd.date_range("2000", periods=5, freq="7D").values.astype("datetime64[us]")
+    cftimes = cftime_range("2000", periods=5, freq="7D", calendar="standard").values
+
+    offset = np.datetime64("0001-01-01")
+    cfoffset = cftime.DatetimeGregorian(1, 1, 1)
+
+    result = duck_array_ops.datetime_to_numeric(
+        times, offset=offset, datetime_unit="D", dtype=int
+    )
+    cfresult = duck_array_ops.datetime_to_numeric(
+        cftimes, offset=cfoffset, datetime_unit="D", dtype=int
+    )
+
+    expected = 730119 + np.arange(0, 35, 7)
+
+    # Outputs are different. Expected ?
```

Comment:
Standard library datetimes and NumPy datetimes both use a proleptic Gregorian calendar (i.e. they assume that "[the current Gregorian calendar always was, and always will be, in effect](https://docs.python.org/3/library/datetime.html#available-types)").  I believe if you use `cftime.DatetimeProlepticGregorian` objects in the cftime-time context that things should match

I realize it's a little counterintuitive that `calendar="standard"` does not match what's typically used in Python; our decision there was based on [what CF conventions dictate](http://cfconventions.org/cf-conventions/cf-conventions.html#calendar).

Reply:
Thanks, done. I thought standard was synonym with proleptic_calendar.